mitodl/ocw-studio
# Cc/collections limited

#### Pre-Flight checklist



#### What are the relevant tickets? #1038 
( a followup to #1048 ) uses https://github.com/mitodl/ocw-hugo-projects/pull/128

#### What's this PR do?
This PR changes the ResourcePickerDialog to show tabs based on the site configuration. With <PROJECT PR GOES HERE> 
- `ocw-course` sites are configured to link resources & pages and embed resources
- `ocw-www` sites are configured to link collections

**Why only link collections in ocw-www?** Because linking to collections in `ocw-www` sites is the immediate need and ... right now there are some mismatches between `ocw-course` and `ocw-www` configurations which would make linking to other things (pages, resources) a bit annoying. For example, in courses, pages have `type=page`, but in www they have `type=pages`. And in courses, resource types are determined by `resourcename` but in `www` they are determined by `filename`.

#### How should this be manually tested?
First, get the new ocw-www and ocw-course site configs:
- for courses: https://github.com/mitodl/ocw-hugo-projects/blob/cc/collections-limited/ocw-course/ocw-studio.yaml
- for www: https://github.com/mitodl/ocw-hugo-projects/blob/cc/collections-limited/ocw-www/ocw-studio.yaml

Then check that:
- within a course, you can link to pages and resources
- within a course, you can embed resources
- within ocw-www sites, you can link to collections
- within ocw-www sites, there is no "embed" option

And maybe check that the course collection links actually render in hugo output, which was an issue with #1048 

#### Screenshots (if appropriate)
<img width="553" alt="Screen Shot 2022-02-24 at 11 24 23 AM" src="https://user-images.githubusercontent.com/9010790/155572162-a60bceb2-593b-411f-8e77-1f91cdaaff34.png">
<img width="606" alt="Screen Shot 2022-02-24 at 11 27 13 AM" src="https://user-images.githubusercontent.com/9010790/155572165-32e05b09-59f4-404d-8c67-a748dd96566c.png">
